waspc/src/Wasp/Project/ExternalConfig/PackageJson.hs

```diff
@@ -0,0 +1,37 @@
+module Wasp.Project.ExternalConfig.PackageJson
+  ( analyzePackageJsonFile,
+    findPackageJsonFile,
+  )
+where
+
+import Control.Monad.Except (ExceptT (ExceptT), runExceptT)
+import qualified Data.Aeson as Aeson
+import Data.Either.Extra (maybeToEither)
+import StrongPath (Abs, Dir, File, Path', toFilePath)
+import qualified Wasp.ExternalConfig.PackageJson as P
+import Wasp.Generator.ExternalConfig.PackageJson (validatePackageJson)
+import Wasp.Project.Common
+  ( PackageJsonFile,
+    WaspProjectDir,
+    findFileInWaspProjectDir,
+    packageJsonInWaspProjectDir,
+  )
+import Wasp.Util (validateToEither)
+import qualified Wasp.Util.IO as IOUtil
+
+analyzePackageJsonFile :: Path' Abs (Dir WaspProjectDir) -> IO (Either [String] P.PackageJson)
+analyzePackageJsonFile waspProjectDir = runExceptT $ do
+  packageJsonFile <- ExceptT findPackageJsonFileOrError
+  packageJson <- ExceptT $ readPackageJsonFile packageJsonFile
+  ExceptT $ validateToEither validatePackageJson packageJson
```

Comment:
I was looking at this call of validateToEither here to figure out if we might not need it, since it seemed to me it is oddly specific function and we could probably replace it with something nicer.

I was looking to do something like following instead:
```hs
ExceptT $ <sometransformation> $ validatePackageJson packageJson
```
So `validatePackageJson packageJson` gives us `IO [String]`. If this array is not empty , we want to shortcircuit the computation with this array sinced it is a list of errors. We could do this with following function:
```hs
\errors -> if null errors then Right () else Left errors
```

I guess we could name it like this:
```hs
errorsListToEither [] = Right ()
errorsListToEither _ = Left errors
```

And then we map it:
```hs
ExceptT $ errorsListToEither <$> validatePackageJson packageJson
```
actually, fmap it, since it doesn't operate in `IO`.

Although, I noticed in another comment that `validatePackageJson packageJson` doesn't even have to be `IO`, so that will also change things here, but I think the idea will remain the same.

Another approach which I have only a rough hint it might work but you can give it a try and see if it makes sense or not, buit I think `ExceptT` might be instance of ErrorMonad (or is it MonadError?), therefore supporting stuff like `throw` (or is it `throw<something>`?) and `try` or something like it. If that is so, you might be able to just do something like
```hs
case validatePackageJson packageJson of
  [] -> pure packageJson
  errors -> throwError errors
```
I assumed here that you made `validatePackageJson` to not be `IO`.

Although `throwError` here you could probably just replace with `ExceptT . pure . Left`, that is what it will be doing in the background I believe.

Reply:
I went with something super simple:

```haskell
case validateTsConfig tsConfig of
  [] -> return tsConfig
  errors -> throwError errors
  ```

Thanks for the tip, it makes sense to inline it.

Reply:
Ok perfect! I wrote a lot above in the comment in case train of thought is useful, but yeah this is the optimal solution at the end.